cpp/src/tsne/tsne_runner.cuh

```diff
@@ -167,7 +168,7 @@ class TSNE_runner {
   {
     distance_and_perplexity();
 
-    const auto NNZ  = COO_Matrix.nnz;
+    const auto NNZ  = (value_idx)COO_Matrix.nnz;
```

Comment:
```suggestion
    const auto NNZ  = value_idx(COO_Matrix.nnz);
```
Let's not let it prevent merge, but if we need to make further changes, it would be good to avoid raw casts.